packages/database-persistence-layer/src/create/create-async-debounce.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Performs a leading edge debounce of async functions.
+ *
+ * If three functions are throttled at the same time:
+ * - The first happens immediately.
+ * - The second is never called.
+ * - The third happens `delayMS` milliseconds after the first has resolved.
+ *
+ * This is distinct from `lodash.debounce` in that it waits for promise
+ * resolution.
+ *
+ * @param {number} delayMS A delay in milliseconds.
+ * @return {Function} A function that debounce whatever function is passed
+ *                    to it.
+ */
+export default function createAsyncDebounce( delayMS ) {
```

Comment:
Just tried that library so that I don't have to write tests, but it seems buggy for a leading edge debounce. It alternates between leading and trailing. I can see why—it never clears the promise after resolution of a leading edge debounce, so it falls through to trailing on every other call.

The package seems unmaintained, so probably no point making an upstream fix, but I made an issue - https://github.com/szchenghuang/debounce-async/issues/5

Reply:
I think I saw another package in npm if you wanted to try that.

Reply:
It was painful, but I managed to write some tests.